src/script/view_model/content/AppLockViewModel.ts

```diff
@@ -46,28 +47,41 @@ export enum APPLOCK_STATE {
 
 const APP_LOCK_STORAGE = 'app_lock';
 
+const getInactivityTimeout = () => {
+  const teamState = container.resolve(TeamState);
+  const appLock = teamState.teamFeatures()['app-lock'];
+  if (appLock.status === 'enabled') {
+    return appLock.config.inactivity_timeout_secs;
+  }
+  return null;
+};
+
 const getTimeout = (queryName: string, configName: 'APPLOCK_SCHEDULED_TIMEOUT' | 'APPLOCK_UNFOCUS_TIMEOUT') => {
   const queryTimeout = parseInt(UrlUtil.getURLParameter(queryName), 10);
+  const backendTimeout = getInactivityTimeout();
   const configTimeout = Config.getConfig().FEATURE && Config.getConfig().FEATURE[configName];
-  const isNotFinite = (value: number) => !Number.isFinite(value);
-  if (isNotFinite(queryTimeout) && isNotFinite(configTimeout)) {
-    return null;
+
+  console.info('###', queryTimeout, configTimeout, backendTimeout);
+
+  if (Number.isFinite(queryTimeout)) {
+    return queryTimeout;
   }
-  if (isNotFinite(queryTimeout)) {
-    return configTimeout;
+  if (Number.isFinite(backendTimeout)) {
+    return backendTimeout;
   }
-  if (isNotFinite(configTimeout)) {
-    return queryTimeout;
+  if (Number.isFinite(configTimeout)) {
+    return configTimeout;
   }
-  return Math.min(queryTimeout, configTimeout);
+  return null;
 };
 
-const getUnfocusAppLockTimeoutInMillis = () => getTimeout(QUERY_KEY.APPLOCK_UNFOCUS_TIMEOUT, 'APPLOCK_UNFOCUS_TIMEOUT');
-const getScheduledAppLockTimeoutInMillis = () =>
+const getUnfocusAppLockTimeoutInSeconds = () =>
+  getTimeout(QUERY_KEY.APPLOCK_UNFOCUS_TIMEOUT, 'APPLOCK_UNFOCUS_TIMEOUT');
+const getScheduledAppLockTimeoutInSeconds = () =>
   getTimeout(QUERY_KEY.APPLOCK_SCHEDULED_TIMEOUT, 'APPLOCK_SCHEDULED_TIMEOUT');
 
-const isUnfocusAppLockEnabled = () => getUnfocusAppLockTimeoutInMillis() !== null;
-const isScheduledAppLockEnabled = () => getScheduledAppLockTimeoutInMillis() !== null;
+const isUnfocusAppLockEnabled = () => getUnfocusAppLockTimeoutInSeconds() !== null;
+const isScheduledAppLockEnabled = () => getScheduledAppLockTimeoutInSeconds() !== null;
 
 export const isAppLockEnabled = () => isUnfocusAppLockEnabled() || isScheduledAppLockEnabled();
```

Comment:
Depending on the spec we might want to adjust this condition. It's possible that we'll look on the feature `status` from the backend.